src/s3-storage-provider.js

```diff
@@ -13,9 +13,17 @@ let _ = require('lodash');
  * wrapper.  Maybe we should consider adding this wrapper to the
  * aws-sdk class...
  */
-let wrapSend = (upload) => {
+let wrapSend = (upload, stream) => {
   return new Promise((res, rej) => {
     // TODO: Make this configurable?
+    stream.on('aborted', () => {
+      upload.abort();
+      console.log('upload aborted because of http/https aborted event');
+    });
+    stream.on('error', err => {
+      upload.abort();
+      console.log(err.stack || err);
+    });
     let abortTimer = setTimeout(upload.abort.bind(upload), 1000 * 60 * 60);
```

Comment:
While I agree with the sentiment, I suspect that each backend is going to have to implement this completely differently.  I think passing the input stream, and maybe note that handling the error and aborted events is not optional.